Tentative implementation of the TryFrom trait

vigna opened the pull request:
This is my first try. I decided to implement TryFrom rather than From, since all `make` methods return an error. Presently I have implemented conversions towards Mmap and MmapMut from Mmap, MmapMut and MmapNone. I have added a reference blurb to the documentation of the `make` methods.

How does that look?

Would it make sense to have these four TryFrom also for the `Reserved*` structs?

StephanvanSchaik commented:
> How does that look?

Re-using the the existing `make_*` functions to just take the `Error` out and drop the `Mmap` object looks good to me.

> Would it make sense to have these four TryFrom also for the Reserved* structs?

Yes, that would make it complete and consistent.

vigna commented:
I added the remaining TryFrom. I also converted a few existing TryInto into TryFrom, because the latter is more generic (there's a blanket for TryInto).

StephanvanSchaik commented:
Sure! Let me look at fixing some minor issues (e.g. FreeBSD CI and the MSRV PR) and then I will prepare a release that incorporates this PR.
